src/Integration/Binding/BindingWorkflow.cs

```diff
@@ -207,8 +206,17 @@ private ProgressStepDefinition[] CreateWorkflowSteps(IProgressController control
             foreach (var language in languages)
             {
                 notifier.NotifyCurrentProgress(string.Format(CultureInfo.CurrentCulture, Strings.DownloadingQualityProfileProgressMessage, language.Name));
+
+                QualityProfile qualityProfileInfo;
+                if (!host.SonarQubeService.TryGetQualityProfile(this.connectionInformation, this.project, language.ServerKey, cancellationToken, out qualityProfileInfo))
+                {
+                    failed = true;
+                    break;
```

Comment:
That seems like the wrong behaviour and should be documented at least with a TODO/issue.
The code purports to support multiple languages: it should either do the right thing or not pretend to handle anything other than C#.